src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -258,6 +261,66 @@ public function downloadVirtualFileAction(int $virtualProductFileId): BinaryFile
         return $response;
     }
 
+    /**
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param string $languageCode
+     *
+     * @return JsonResponse
+     */
+    public function searchAssociationsAction(Request $request, string $languageCode): JsonResponse
+    {
+        $langRepository = $this->get('prestashop.core.admin.lang.repository');
+        $lang = $langRepository->getOneByLocaleOrIsoCode($languageCode);
+        if (null === $lang) {
+            return $this->json([
+                'message' => sprintf(
+                    'Invalid language code %s was used which matches no existing language in this shop.',
+                    $languageCode
+                ),
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        $shopId = (int) $this->get('prestashop.adapter.shop.context')->getContextShopID();
+        if (empty($shopId)) {
```

Comment:
```suggestion
        if (0 === $shopId) {
```

Reply:
No `empty` because it can return `null` (actually it's mostly null that I wan to check, but I'm not 100% sure 0 will never pop up)

Reply:
As you cast it into integer, it can be null

Reply:
here `$shopId` will never be null since it's casted in integer, if the result of `getContextShopID` is null then `$shopId` will be 0